iothub_client/src/iothub_client_diagnostic.c

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdlib.h>
+#include <math.h>
+#include "azure_c_shared_utility/optimize_size.h"
+#include "azure_c_shared_utility/gballoc.h"
+#include "azure_c_shared_utility/xlogging.h"
+#include "azure_c_shared_utility/buffer_.h"
+
+#include "iothub_client_diagnostic.h"
+
+static const int BASE_36 = 36;
+
+static char* get_current_time_utc(char* timeBuffer, int bufferLen, const char* formatString)
+{
+    time_t epochTime;
+    struct tm* utcTime;
+    epochTime = time(NULL);
+    utcTime = gmtime(&epochTime);
+    strftime(timeBuffer, bufferLen, formatString, utcTime);
+    return timeBuffer;
+}
+
+static char get_base36_char(unsigned char value)
+{
+    if (value <= 9)
+    {
+        return '0' + value;
+    }
+    else
+    {
+        return 'a' + value - 10;
+    }
+}
+
+static char* generate_eight_random_characters(char *randomString)
```

Comment:
This function doesn't seem to generate true random strings. I believe there is a good chance of collision here.

You should opt to use UUID (RFC 4122). 

There is a module in c-utility (uniqueid.h) that can generate UUID strings.

Reply:
UUID was our first proposal, but changed to 8 digit characters after review